### ISSUE TITLE
Terraform plan does not detect updates when graphs updated outside terraform

### ISSUE THREAD

**Issue by nyanshak** [terraform-providers/terraform-provider-datadog#20]
### Terraform Version
Terraform v0.10.5

### Affected Resource(s)
- datadog_timeboard

### Terraform Configuration Files
```hcl
resource "datadog_timeboard" "my_timeboard" {
  title = "My timeboard"
  description = "My timeboard description"

  graph {
    title = "My Graph Title"
    viz = "timeseries"
    request {
      q = "sum:aws.elb.request_count{*}.as_count()"
      aggregator = "avg"
      type = "line"
    }
  }
}
```
### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` with the above config
2. Go to Datadog's UI and add another request to the graph (doesn't really matter, just any second request).
3. `terraform plan`

### Expected Behavior
Step 3 (terraform plan) should show that the timeboard needs to be updated (to remove the extra unexpected request on the graph.

### Actual Behavior
Terraform plan output shows no changes expected. Terraform apply does not apply any changes.

### Other Notes
I've noticed that terraform WILL correctly detect certain changes (such as title or description updates) but will not detect additional graphs. I'm not sure if there are OTHER things that aren't causing plan to report updates.

**Comment by yfronto** [comment]
The `schema.ResourceData` isn't getting populated. Things are failing but errors are being silently ignored. In the `resourceDatadogTimeboardRead` function, the call to `schema.ResourceData.Set` [fails but isn't caught][1]. If I make the change to check errors, the failures are there and the errors vary with each run:

```
k@Thor ~/test> terraform import datadog_timeboard.my_timeboard 361820
datadog_timeboard.my_timeboard: Importing from ID "361820"...
Error importing: 1 error(s) occurred:

* datadog_timeboard.my_timeboard (import id: 361820): import datadog_timeboard.my_timeboard (id: 361820): Invalid address to set: []string{"graph", "0", "group"}
k@Thor ~/test [1]> terraform import datadog_timeboard.my_timeboard 361820
datadog_timeboard.my_timeboard: Importing from ID "361820"...
Error importing: 1 error(s) occurred:

* datadog_timeboard.my_timeboard (import id: 361820): import datadog_timeboard.my_timeboard (id: 361820): Invalid address to set: []string{"graph", "0", "events"}
k@Thor ~/test [1]> terraform import datadog_timeboard.my_timeboard 361820
datadog_timeboard.my_timeboard: Importing from ID "361820"...
Error importing: 1 error(s) occurred:

* datadog_timeboard.my_timeboard (import id: 361820): import datadog_timeboard.my_timeboard (id: 361820): graph.0.include_ungrouped_hosts: '' expected type 'bool', got unconvertible type '*bool'
```
My guess is that it has to do with the refactor of the client library to generate its accessor methods and move to pointers for all the fields coupled with Terraform's use of the reflect library to anticipate a need for dereference.

The solution isn't immediately obvious to me but I can take another look at it in a while.

[1]: https://github.com/terraform-providers/terraform-provider-datadog/blob/d4b68e5e366d3f1081ec9285ce9213be57be7b40/datadog/resource_datadog_timeboard.go#L656

**Comment by nyanshak** [comment]
I believe this was fixed in #27. Can close this once it gets added to a release.
